modules/oe_webtools_analytics/modules/oe_webtools_analytics_rules/src/RuleMatcher.php

```diff
@@ -0,0 +1,273 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_webtools_analytics_rules;
+
+use Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException;
+use Drupal\Component\Plugin\Exception\PluginNotFoundException;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Path\AliasManagerInterface;
+use Drupal\Core\Path\CurrentPathStack;
+use Drupal\oe_webtools_analytics_rules\Entity\WebtoolsAnalyticsRule;
+use Drupal\oe_webtools_analytics_rules\Entity\WebtoolsAnalyticsRuleInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Service that matches routes to rules.
+ */
+class RuleMatcher implements RuleMatcherInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
```

Comment:
Nit: can we sort alphabetically the properties? 😛

Reply:
Oh yiss :smile_cat: I'll even inject them alphabetically :laughing: